tests/recipes/cad1/task1/test_enhance.py

```diff
@@ -0,0 +1,253 @@
+                [2.44948974, 3.67423461, 4.89897949],
+            ]
+        ),
+        np.array([3.0, 4.0, 5.0]),
+    )
+    result = normalize_signal(signal)
+    assert len(result) == len(expected_result)
+    assert np.allclose(result[0], expected_result[0])
+    assert np.allclose(result[1], expected_result[1])
+
+
+def test_denormalize_signals():
+    # Test case 1: Test the function with a simple input
+    sources = np.array(
+        [
+            np.array(
+                [
+                    [-1.22474487, 0.0, 1.22474487],
+                    [-1.22474487, 0.0, 1.22474487],
+                    [-1.22474487, 0.0, 1.22474487],
+                ]
+            ),
+            np.array([[0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0]]),
+        ]
+    )
+    ref = np.array([4.0, 5.0, 6.0])
+    expected_result = [
+        np.array([[4.0, 5.0, 6.0], [4.0, 5.0, 6.0], [4.0, 5.0, 6.0]]),
+        np.array([[5.0, 5.0, 5.0], [5.0, 5.0, 5.0], [5.0, 5.0, 5.0]]),
+    ]
+    result = denormalize_signals(sources, ref)
+    assert len(result) == len(expected_result)
+    assert np.allclose(result[0], expected_result[0])
+    assert np.allclose(result[1], expected_result[1])
+
+    # Test case 2: Test the function with an input that has a non-zero mean
+    sources = np.array(
+        [
+            np.array(
+                [
+                    [-1.22474487, 0.0, 1.22474487],
+                    [-1.22474487, 0.0, 1.22474487],
+                    [-1.22474487, 0.0, 1.22474487],
+                ]
+            ),
+            np.array([[0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0]]),
+        ]
+    )
+    ref = np.array([1.0, 2.0, 3.0])
+    expected_result = [
+        np.array([[1.0, 2.0, 3.0], [1.0, 2.0, 3.0], [1.0, 2.0, 3.0]]),
+        np.array([[2.0, 2.0, 2.0], [2.0, 2.0, 2.0], [2.0, 2.0, 2.0]]),
+    ]
+    result = denormalize_signals(sources, ref)
+    assert len(result) == len(expected_result)
+    assert np.allclose(result[0], expected_result[0])
+    assert np.allclose(result[1], expected_result[1])
+
+    # Test case 3: Test the function with an input that has a non-unit variance
+    sources = np.array(
+        [
+            np.array(
+                [
+                    [-2.44948974, 0.0, 2.44948974],
+                    [-2.44948974, 0.0, 2.44948974],
+                    [-2.44948974, 0.0, 2.44948974],
+                ]
+            ),
+            np.array([[0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0]]),
+        ]
+    )
+    ref = np.array([4.0, 5.0, 6.0])
+    expected_result = [
+        np.array([[3.0, 5.0, 7.0], [3.0, 5.0, 7.0], [3.0, 5.0, 7.0]]),
+        np.array([[5.0, 5.0, 5.0], [5.0, 5.0, 5.0], [5.0, 5.0, 5.0]]),
+    ]
+    result = denormalize_signals(sources, ref)
+    assert len(result) == len(expected_result)
+    assert np.allclose(result[0], expected_result[0])
+    assert np.allclose(result[1], expected_result[1])
+
+
+def test_map_to_dict(regtest):
```

Comment:
I don't think this really needs to be a `regtest` you can just define the expected dictionary and `assert` the returned object is the same.